src/mito2/src/region/opener.rs

```diff
@@ -124,20 +133,72 @@ impl RegionOpener {
             }
         );
 
+        let region_id = metadata.region_id;
         let mutable = self.memtable_builder.build(&metadata);
         let version = VersionBuilder::new(metadata, mutable).build();
-        let version_control = Arc::new(VersionControl::new(version));
+        let flushed_sequence = version.flushed_sequence;
 
-        // TODO(yingwen): Replay.
+        let version_control = Arc::new(VersionControl::new(version));
+        replay_memtable(
+            &self.wal,
+            region_id,
+            flushed_sequence,
+            version_control.clone(),
+        )
+        .await?;
 
-        Ok(MitoRegion {
+        let region = MitoRegion {
             region_id: self.region_id,
             version_control,
             manifest_manager,
-        })
+        };
+        Ok(region)
     }
 }
 
+/// Replays the mutations from WAL and inserts mutations to memtable of given region.
+async fn replay_memtable<S: LogStore>(
+    wal: &Wal<S>,
+    region_id: RegionId,
+    flushed_sequence: SequenceNumber,
+    version_control: VersionControlRef,
+) -> Result<()> {
+    let mut notify_waiter = vec![];
```

Comment:
We don't need to notify waiters during replay.

Reply:
> We don't need to notify waiters during replay.

but we have to wait until all entries' inserted to memtable during replay.